### PR TITLE
Tickets/dm 30177

### PR DESCRIPTION
**Pull request by brendal4**
Added simple file filtering mechanism for handoff manager

### CODE REVIEW

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 from datetime import datetime
 from .abcs import Command
 from .messages import FileMsg
+import re
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:30]: By convention (see [PEP8](https://pep8.org/#imports)), imports of Python standard libraries (like **re**) should go first. Also, please separate the imports of the standard libraries from the imports of the local modules by a blank line.

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 2]
```diff
@@ -37,14 +38,12 @@
 
 class Finder(Command):
     """Command finding out all files in the buffer on the handoff site.
-
     Parameters
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:41]: Please put back the blank line you removed; it's not optional (follow this [link](https://developer.lsst.io/python/numpydoc.html#basic-format-of-docstrings) for details).

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 3]
```diff
@@ -37,14 +38,12 @@
...
     ----------
     config : dict
         Configuration of the handoff site.
     queue : queue.Queue
         Container where the files found in the given directory will be stored.
-
     Raises
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:47]: See the comment above.

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 4]
```diff
@@ -60,15 +59,28 @@ def __init__(self, config, queue):
...
+        self.exclude_list = config.get("exclude_list")
+        if self.exclude_list is None:
+            self.exclude_list = []
 
     def run(self):
         """Scan recursively the directory to find all files it contains.
         """
+        
+        
         for topdir, _, filenames in os.walk(self.root):
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:71]: Remove this two blank lines.

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 5]
```diff
@@ -60,15 +59,28 @@ def __init__(self, config, queue):
...
             for name in filenames:
                 path = os.path.join(topdir, name)
                 tail = os.path.relpath(path, start=self.root)
                 dirname, basename = os.path.split(tail)
+                
+                matches = [f"'{patt}'" for patt in self.exclude_list                     # <--  This is a list called "matches" which picks out each pattern (called "patt") in "exclude_list" 
+                       if re.search(patt, tail) is not None]
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:78]: Fix the indentation to:
```python
matches = [f"'{patt}'" for patt in self.exclude_list
           if re.search(patt, tail) is not None]
```

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 6]
```diff
@@ -60,15 +59,28 @@ def __init__(self, config, queue):
...
         """
+        
+        
         for topdir, _, filenames in os.walk(self.root):
             for name in filenames:
                 path = os.path.join(topdir, name)
                 tail = os.path.relpath(path, start=self.root)
                 dirname, basename = os.path.split(tail)
+                
+                matches = [f"'{patt}'" for patt in self.exclude_list                     # <--  This is a list called "matches" which picks out each pattern (called "patt") in "exclude_list"
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:77]: Please remove this annotation (and the ones that follow).

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 7]
```diff
@@ -85,7 +97,6 @@ def run(self):
 
 class Mover(Command):
     """Command moving files between the buffer and the holding area.
-
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:88]: Put the blank line back.

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 8]
```diff
@@ -94,7 +105,6 @@ class Mover(Command):
         Input message queue with files to move.
     out : queue.Queue
         Output message queue with files that were moved.
-
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:97]: See the earlier comment.

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 9]
```diff
@@ -138,19 +148,16 @@ def run(self):
 
 class Eraser(Command):
     """Command removing empty directories from the buffer.
-
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:141]: As above.

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 10]
```diff
@@ -138,19 +148,16 @@ def run(self):
...
     To avoid possible race condition between the application writing files to
     the buffer and the command itself, empty directories are removed only if
     they were not modified for a certain period of time.
-
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:145]: As above.

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 11]
```diff
@@ -138,19 +148,16 @@ def run(self):
...
     Parameters
     ----------
     config : dict
         Configuration of the handoff site.
     exp_time : int
         Time (in seconds) that need to pass from the last modification before
         an empty directory can be removed.
-
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:153]: As above.

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 12]
```diff
@@ -60,15 +59,28 @@ def __init__(self, config, queue):
...
                 tail = os.path.relpath(path, start=self.root)
                 dirname, basename = os.path.split(tail)
+                
+                matches = [f"'{patt}'" for patt in self.exclude_list                     # <--  This is a list called "matches" which picks out each pattern (called "patt") in "exclude_list" 
+                       if re.search(patt, tail) is not None]
+                if matches:                                                         # <--  Here, if the "matches" list isn't empty, "logger.debug" runs which prints as a message the
+                    logger.debug("%s was excluded by pattern(s): "                  #      name of the path (with filename appended) that was skipped
+                                 "%s", path, ', '.join(matches))
+                    continue                 
+
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:83]: Please remove the annotations.

File: python/lsst/dbb/buffmngrs/handoff/local.py [hunk 13]
```diff
@@ -194,3 +201,5 @@ def run(self):
                     logger.warning(f"Cannot remove '{path}': {ex}")
                 else:
                     logger.debug(f"Directory '{path}' removed successfully.")
+                    
+
```
Comment (mxk62) [inline python/lsst/dbb/buffmngrs/handoff/local.py:205]: Remove these two blank lines. Make sure that the last line ends with newline character though.

Comment (timj) [reply python/lsst/dbb/buffmngrs/handoff/local.py:205]: I think you need to turn on the standard lint github action so that these are all spotted early.

**Pull request closed without merging**